src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -194,15 +204,100 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
+        this.dropdownClickNumber = 0;
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen click event whether on outside of the menu or one of the items in the menu to close the dropdown menu */
+    @HostListener('document: click', ['$event'])
+    onClickCloseDropdownMenu() {
+        if (this.dropdownOpen) {
+            this.dropdownClickNumber += 1;
+            if (this.dropdownClickNumber === 2) {
+                this.dropdownOpen = false;
+                this.dropdownClickNumber = 0;
+            }
+        }
+    }
+
+    /** Update sidebar item's hidden property based on the window height to display three-dots */
+    updateVisibility(height: number) {
+        const thresholds: number[] = [];
+        const itemHeight: number = 35;
+        let threshold = this.calculateThreshold();
+        this.anyItemHidden = false;
+        this.hiddenItems = [];
+
+        for (let i = 0; i < this.sidebarItems.length - 1; i++) {
+            thresholds.unshift(threshold);
+            threshold -= itemHeight;
+        }
+        thresholds.unshift(0);
+
+        this.sidebarItems.forEach((item, index) => {
+            item.hidden = height <= thresholds[index];
+            if (item.hidden) {
+                this.anyItemHidden = true;
+                this.hiddenItems.push(item);
+            }
+        });
+    }
+
+    /** Calculate dropdown-menu position based on the number of entries in the sidebar */
+    updateMenuOffset() {
+        const leftSidebarItems: number = this.sidebarItems.length - this.hiddenItems.length;
+        if (leftSidebarItems === 1) {
+            this.dropdownOffset = 0;
+            if (window.innerHeight > 370) {
+                this.dropdownOffset += (9 - this.hiddenItems.length) * 10;
+            }
+        } else {
+            this.dropdownOffset = leftSidebarItems * 50;
+        }
+    }
+
+    /** Calculating threshold levels based on the number of entries in the sidebar */
+    calculateThreshold() {
+        const maxThreshold: number = 650;
+        const numberOfSidebarItems: number = this.sidebarItems.length;
+        if (numberOfSidebarItems === 9) {
+            return maxThreshold;
+        }
+        return maxThreshold - (9 - numberOfSidebarItems) * 35;
+    }
+
+    toggleDropdown() {
+        this.dropdownOpen = !this.dropdownOpen;
+        // Refresh click numbers after toggle
+        if (!this.dropdownOpen) {
+            this.dropdownClickNumber = 0;
+        }
+    }
+
+    dropdownOffsetToString() {
+        return `${this.dropdownOffset}px`;
     }
 
     getCourseActionItems(): CourseActionItem[] {
         const courseActionItems = [];
+        /*
         this.canUnenroll = this.canStudentUnenroll() && !this.course?.isAtLeastTutor;
         if (this.canUnenroll) {
             const unenrollItem: CourseActionItem = this.getUnenrollItem();
             courseActionItems.push(unenrollItem);
         }
+        */
```

Comment:
This was for testing purposes, I assume - please add it back.

Reply:
You are right, sorry I was trying something and forgot to uncomment! Thanks for detecting it